hoymiles_modbus/client.py

```diff
@@ -43,9 +41,10 @@ def _data_length_fixer(packet):  # pragma: no cover
             fixed_packet[8] = len(fixed_packet[9:])
         return bytes(fixed_packet)
 
-    def processIncomingPacket(self, data, callback, unit, **kwargs):
```

Comment:
processIncomingPacket() still exists in the legacy framer so there should be no need to change it to decode(). Have you verified this on your setup?

Reply:
Yes. During runtime, `processIncomingPacket()` won't be called, `decode()` will.

I'm not changing the implementation just for the fun of making changes. I traversed that path and it did not yield the result. This is the minimum change that will actually run, operate and work.

Reply:
I'm sorry but your code doesn't work for me:

`2024-08-12 22:05:02,988 [Thread-1 (ex] [ERROR]  [pymodbus.logging] Unable to decode response Modbus Error: [Input/Output] Invalid response b')\x0c\x10!b\x81cU\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x08\xb6\x00\x02\x1b\x86\x00\x00\x00\x00\x00\x00\x00\x00\x00\x07\x00\x00\x00\x00\x00\x00' has byte count of 41`

It started working when I replaced `decode()` method with the following:

```
    def processIncomingPacket(self, data: bytes, callback, slave, single=False, tid=None):
        fixed_data = self._data_length_fixer(data)
        return super().processIncomingPacket(fixed_data, callback, slave, single, tid)
```

Reply:
If the code is working for you then perhaps your DTU doesn't require the workaround.